core/src/main/scala/org/apache/spark/sql/rapids/tool/util/RapidsToolsConfUtil.scala

```diff
@@ -18,62 +18,53 @@ package org.apache.spark.sql.rapids.tool.util
 
 import org.apache.hadoop.conf.Configuration
 
-import org.apache.spark.SparkConf
-import org.apache.spark.deploy.SparkHadoopUtil
 import org.apache.spark.internal.Logging
 import org.apache.spark.sql.SparkSession
 
 /**
  * Contains util methods to interact with Spark/Hadoop configurations.
  */
-
 object RapidsToolsConfUtil extends Logging {
-  val hadoopConfPrefix = s"${RAPIDS_TOOLS_SYS_PROP_PREFIX}hadoop."
-  val sparkConfPrefix = s"${RAPIDS_TOOLS_SYS_PROP_PREFIX}spark."
-
-  private lazy val configMap = loadConfFromSystemProperties
+  private val HADOOP_CONF_PREFIX = s"${RAPIDS_TOOLS_SYS_PROP_PREFIX}hadoop."
 
   /**
-   * Creates a sparkConfiguration object from the existing sparkSession if any.
-   * Then it will call {@link RapidsToolsConfUtil.newHadoopConf(org.apache.spark.SparkConf)}
+   * Creates a sparkConfiguration object with system properties applied on-top.
    * @return a hadoop configuration object
    */
   def newHadoopConf(): Configuration = {
-    val sparkConf = SparkSession.getActiveSession match {
-      case Some(spark) => spark.sparkContext.getConf
-      case None => new SparkConf()
-    }
-    newHadoopConf(sparkConf)
+    newHadoopConf(loadConfFromSystemProperties)
   }
 
   /**
-   * Returns a Configuration object with Spark configuration applied on top.
-   * It also applies system properties with prefix "rapids.tools.hadoop" on top of all.
-   *
-   * @param conf a spark configuration instance
+   * Created a new Configuration object after applying the properties on top.
+   * The configuration map keys should be prefixed with "rapids.tools.hadoop."
```

Comment:
@tgravescs - It would be good to get your perspective on this. Is it okay to change the hadoop configs with above prefix? This prefix is removed later on in the code.

Reply:
To clarify the motivation behind this
While Hadoop allows user to run Hadoop commands with "-D prop=xyz", there is no interface to do that within java application.
The current way to do that is to create configuration files that can be loaded by the java application.

As we focus on running the tools in "local" mode (does not require spark/hadoop installations); we need to offer a convenient way to pass hadoop configurations without need to create configuration folders.
This PR provides that feature allowing to pass the connection strings as `-Drapids.tools.hadoop.*=value`.

Reply:
right, spark does essentially the same thing with spark.hadoop.fs.s3a.access.key, they looks for spark.hadoop and then put those into the Configuration